### ISSUE TITLE
pmr::monotonic_buffer_resource crashes on large single allocations

### ISSUE THREAD

**Issue by jmlundberg** [boostorg/container#159]
```
boost::container::pmr::monotonic_buffer_resource mr (boost::container::pmr::get_default_resource());
boost::container::pmr::vector<char> vchar2(&r);
const auto zz = 5368709127;
std::cout << zz <<std::endl;
vchar.resize(zz);
```
crashes at memset. windows, msvc, 64 bit.

**Comment by vinniefalco** [comment]
The function `increase_next_buffer_at_least_to`  here:
https://github.com/boostorg/container/blob/5a52472cd00994bf1752e92c1c178a5a822a36cb/src/monotonic_buffer_resource.cpp#L148

Does not make the next buffer large enough to hold `bytes`, because the math function `floor_log2` it relies on is 32-bit and not 64-bit:
https://github.com/boostorg/intrusive/blob/d8761780b0386c53ab144173a5fdc536ee2e1da8/include/boost/intrusive/detail/math.hpp#L79

**Comment by igaztanaga** [comment]
Many thanks for the report. As Vinnie has correctly identified, it's a Boost.Intrusive problem, just fixed in boostorg/intrusive#52.
